mobile/lib/models/selected_albums.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:collection/collection.dart' show IterableExtension;
+import 'package:flutter/foundation.dart';
+import 'package:photos/core/event_bus.dart';
+import "package:photos/events/clear_album_selections_event.dart";
+import 'package:photos/models/collection/collection.dart';
+
+class SelectedAlbums extends ChangeNotifier {
+  final albums = <Collection>{};
+
+  void toggleSelection(Collection albumToToggle) {
+    final Collection? alreadySelected = albums.firstWhereOrNull(
+      (element) => element.id == albumToToggle.id,
+    );
+    if (alreadySelected != null) {
+      albums.remove(alreadySelected);
+    } else {
+      albums.add(albumToToggle);
+    }
+    notifyListeners();
+  }
+
+  void selectAll(Set<Collection> albumsToSelect) {
+    albums.addAll(albumsToSelect);
+    notifyListeners();
+  }
+
+  void unSelectAll(
```

Comment:
I feel just `unSelect` would be a better name. At first glance just looking at the name, I expected that calling `SelectedAlbum.unSelectAll()` would just clear all selection. I understand you took the names from `SelectedFiles` but `unSelect` is definitely better.